apps/sched/interface.html

```diff
@@ -88,29 +88,62 @@
 function upload() {
   Util.showModal("Saving...");
   Util.writeStorage("sched.json", JSON.stringify(alarms), () => {
-    location.reload(); // reload so we see current data
+    Puck.write(`\x10load()\n`, () => { // reload watch to load alarms
+      setTimeout(() => {
+        location.reload(); // reload so we see current data
+      }, 500);
+    });
   });
 }
 
 function renderAlarm(alarm, exists) {
-  const localDate = dateFromAlarm(alarm);
+  const localDate = alarm.date ? dateFromAlarm(alarm) : null;
 
   const tr = document.createElement('tr');
   tr.classList.add('event-row');
   tr.dataset.uid = alarm.id;
-  const tdTime = document.createElement('td');
-  tr.appendChild(tdTime);
+  const tdType = document.createElement('td');
+  tdType.type = "text";
+  tdType.classList.add('event-summary');
+  tr.appendChild(tdType);
   const inputTime = document.createElement('input');
-  inputTime.type = "datetime-local";
+  if (localDate) {
+    tdType.textContent = "Alarm";
+    inputTime.type = "datetime-local";
+    inputTime.value = localDate.toISOString().slice(0,16);
+    inputTime.onchange = (e => {
+      const date = new Date(inputTime.value);
+      alarm.t = dateToMsSinceMidnight(date);
+      alarm.date = formatDate(date);
+    });
+  } else {
+    const [hours, mins, secs] = msToHMS(alarm.timer || alarm.t);
+
+    inputTime.type = "time";
+    inputTime.step = 1; // display seconds
+    inputTime.value = `${hours}:${mins}:${secs}`;
+
+    if (alarm.timer) {
+      tdType.textContent = "Timer";
+      inputTime.onchange = e => {
+        alarm.timer = hmsToMs(inputTime.value);
+        const now = new Date();
+        const currentTime = (now.getHours()*3600000)+(now.getMinutes()*60000)+(now.getSeconds()*1000);
+        alarm.t = currentTime + alarm.timer;
+      };
+    } else {
+      tdType.textContent = "Alarm";
```

Comment:
Can you also differentiate beween "Alarm" and
"Event" please? Something like
`alarm.date ? "Event" : "Alarm"`

Reply:
Sounds good - I had been using the different input (datetime vs time) to hint to the user but this is much more clear